network/p2p/subscription_filter_test.go

```diff
@@ -0,0 +1,119 @@
+	t.Skip()
+
+	identity1, privateKey1 := createID(t, unittest.WithRole(flow.RoleAccess))
+	identity2, privateKey2 := createID(t, unittest.WithRole(flow.RoleAccess))
+	ids := flow.IdentityList{identity1, identity2}
+
+	node1 := createNode(t, identity1.NodeID, privateKey1, createSubscriptionFilterPubsubOption(t, ids))
+	node2 := createNode(t, identity2.NodeID, privateKey2, createSubscriptionFilterPubsubOption(t, ids))
+
+	unstakedKey, err := unittest.NetworkingKey()
+	require.NoError(t, err)
+	unstakedNode := createNode(t, flow.ZeroID, unstakedKey)
+
+	require.NoError(t, node1.AddPeer(context.TODO(), *host.InfoFromHost(node2.Host())))
+	require.NoError(t, node1.AddPeer(context.TODO(), *host.InfoFromHost(unstakedNode.Host())))
+
+	badTopic := engine.TopicFromChannel(engine.SyncCommittee, rootBlockID)
+
+	sub1, err := node1.Subscribe(context.TODO(), badTopic)
+	require.NoError(t, err)
+
+	sub2, err := node2.Subscribe(context.TODO(), badTopic)
+	require.NoError(t, err)
+
+	unstakedSub, err := unstakedNode.Subscribe(context.TODO(), badTopic)
+	require.NoError(t, err)
+
+	require.Eventually(t, func() bool {
+		return len(node1.pubSub.ListPeers(badTopic.String())) > 0 &&
+			len(node2.pubSub.ListPeers(badTopic.String())) > 0 &&
+			len(unstakedNode.pubSub.ListPeers(badTopic.String())) > 0
+	}, 1*time.Second, 100*time.Millisecond)
+
+	require.Never(t, func() bool {
+		for _, pid := range node1.pubSub.ListPeers(badTopic.String()) {
+			if pid == unstakedNode.Host().ID() {
+				return true
+			}
+		}
+		return false
+	}, 1*time.Second, 100*time.Millisecond)
+
+	var wg sync.WaitGroup
+	wg.Add(2)
+
+	testPublish := func(wg *sync.WaitGroup, from *Node, sub *pubsub.Subscription) {
+		data := []byte("hello")
+
+		from.Publish(context.TODO(), badTopic, data)
+
+		ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+		msg, err := sub.Next(ctx)
+		cancel()
+		require.NoError(t, err)
+		require.Equal(t, msg.Data, data)
+
+		ctx, cancel = context.WithTimeout(context.Background(), time.Second)
+		msg, err = unstakedSub.Next(ctx)
+		cancel()
+		require.ErrorIs(t, err, context.DeadlineExceeded)
+
+		wg.Done()
+	}
+
+	// publish a message from node 1 and check that only node2 receives
+	testPublish(&wg, node1, sub2)
+
+	// publish a message from node 2 and check that only node1 receives
+	testPublish(&wg, node2, sub1)
+
+	wg.Wait()
+}
+
+func TestCanSubscribe(t *testing.T) {
+	identity, privateKey := createID(t, unittest.WithRole(flow.RoleCollection))
+
+	collectionNode := createNode(t, identity.NodeID, privateKey, createSubscriptionFilterPubsubOption(t, flow.IdentityList{identity}))
+	defer collectionNode.Stop()
+
+	goodTopic := engine.TopicFromChannel(engine.ProvideCollections, rootBlockID)
+	_, err := collectionNode.pubSub.Join(goodTopic.String())
+	require.NoError(t, err)
+
+	badTopic := engine.TopicFromChannel(engine.ProvideReceiptsByBlockID, rootBlockID)
+	_, err = collectionNode.pubSub.Join(badTopic.String())
+	require.Error(t, err)
+
```

Comment:
so, we're validating the `channelRoleMap` at the same time, is that correct?

Reply:
Not exactly, what do you mean by this?